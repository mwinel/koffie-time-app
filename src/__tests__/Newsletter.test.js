import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import Newsletter from "../components/NewsLetter";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders newsletter component", () => {
  act(() => {
    render(<Newsletter />, container);
  });
  expect(container.innerHTML).toBeDefined();
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
