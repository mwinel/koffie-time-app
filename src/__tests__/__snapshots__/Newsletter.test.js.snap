// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders newsletter component 1`] = `
"<section class=\\"newsletter-section my-5\\">
  <div class=\\"container py-5\\">
    <div class=\\"row\\">
      <div class=\\"col-sm-3\\"></div>
      <div class=\\"col-sm-6\\">
        <div class=\\"py-5 my-5\\">
          <h4 class=\\"pb-4\\">Join our newsletter for the latest tips and tricks to grow your career.</h4>
          <form>
            <div class=\\"form-group\\"><input type=\\"email\\" class=\\"form-control\\" placeholder=\\"Email Address\\"></div>
            <div class=\\"form-row\\">
              <div class=\\"form-group col-md-6\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"First name\\"></div>
              <div class=\\"form-group col-md-6\\"><input type=\\"text\\" class=\\"form-control\\" placeholder=\\"Last name\\"></div>
            </div><button type=\\"submit\\" class=\\"sc-bdVaJa gKnoUA btn btn-primary\\">Subscribe Now</button>
          </form>
        </div>
      </div>
    </div>
  </div>
</section>"
`;
